src/intentManager.js

```diff
@@ -35,6 +37,44 @@ define([
         };
     }
 
+    function intentAppIdWidget(mug, options) {
+        var widget = widgets.text(mug, options),
+            input = widget.input,
+            opts = options.vellum.opts().intents,
+            tempMenu = $(intent_templates({templates: opts && opts.templates || []})),
+            templates = _.object(_.map(opts && opts.templates, function (temp) {
+                return [temp.id, temp];
+            })),
+            control = $('<div class="control-row row" />')
+                .append($('<div class="span8" />').append(input))
+                .append($('<div class="span4" />').append(tempMenu));
+
+        tempMenu.find('.dropdown-menu a').click(function (e) {
```

Comment:
Can't easily reason what this block is doing, can add a comment or enclose it in a smaller function? (#not-blocker)